hnn_core/drives.py

```diff
@@ -146,7 +141,8 @@ def _add_drives_from_params(net):
             t0=bias_specs['tonic'][cellname]['t0'],
             tstop=bias_specs['tonic'][cellname]['tstop'])
 
-    net._instantiate_drives(n_trials=net._params['N_trials'])
+    net._instantiate_drives(tstop=net._params['tstop'],
```

Comment:
I understand that we need to preserve the params dict for functions like `add_drives_from_params()`, but I don't understand why we need to reference the params dict in other places.

Do we even need `_instantiate_drives()` in `add_drives_from_params()` since the drives are instantiated again in `simulate_dipole()`?

Reply:
> Do we even need _instantiate_drives() in add_drives_from_params() since the drives are instantiated again in simulate_dipole()?

I tried to remove it but a bunch of tests start failing. Perhaps @cjayb can throw some light!

Reply:
If this will require a bunch more changes, let's save it for later. I was originally thinking it might simplify things but that is obviously not the case.

Reply:
I took a look, and it's not that easy. Several tests, both those using the `_run_hnn_core_fixture` and some not, check that the number of event times, artificial cells, etc. match what is expected based on the parameters. Calling `simulate_dipole` in those checks would make them a lot longer (too long). The instantiation could me moved into the relevant tests, though, but that's a separate PR :)